docs/cody/enterprise/model-config-examples.mdx

```diff
@@ -0,0 +1,649 @@
+# Model Configuration Examples
+
+<p className="subtitle">This section includes examples about how to configure Cody to use Sourcegraph-provided models with `modelConfiguration`. These examples will use the following:</p>
+
+- [Minimal configuration](/cody/enterprise/model-configuration#configure-sourcegraph-provided-models)
+- [Using model filters](/cody/enterprise/model-configuration#model-filters)
+- [Change default models](/cody/enterprise/model-configuration#default-models)
+
+## Sourcegraph-provided models and BYOK (Bring Your Own Key)
+
+By default, Sourcegraph is fully aware of several models from the following providers:
+
+- "anthropic"
+- "google"
+- "fireworks"
+- "mistral"
+- "openai"
+
+### Override configuration of a model provider
+
+Instead of Sourcegraph using its own servers to make LLM requests, it is possible to bring your own API keys for a given model provider. For example, if you wish for all Anthropic API requests to go directly to your own Anthropic account and use your own API keys instead of going via Sourcegraph's servers, you could override the `anthropic` provider's configuration:
+
+```json
+{
+"cody.enabled": true,
+"modelConfiguration": {
+  "sourcegraph": {},
+  "providerOverrides": [
+    {
+     "id": "anthropic",
+    "displayName": "Anthropic BYOK",
+    "serverSideConfig": {
+      "type": "anthropic",
+      "accessToken": "token",
+       "endpoint": "https://api.anthropic.com/v1/messages"
+      }
+    }
+  ],
+  "defaultModels": {
+    "chat": "anthropic::2024-10-22::claude-3.5-sonnet",
+    "fastChat": "anthropic::2023-06-01::claude-3-haiku",
+    "autocomplete": "fireworks::v1::deepseek-coder-v2-lite-base"
+  }
+}
+```
+
+In the configuration above:
+
+- Enable Sourcegraph-provided models and do not set any overrides (note that `"modelConfiguration.modelOverrides"` is not specified)
+- Route requests for Anthropic models directly to the Anthropic API (via the provider override specified for "anthropic")
+- Route requests for other models (such as the Fireworks model for "autocomplete") through Cody Gateway
+
+### Partially override provider config in the namespace
+
+If you want to override the provider config for some models in the namespace and use the Sourcegraph-configured provider config for the rest, you can route requests directly to the LLM provider (bypassing the Cody Gateway) for some models while using the Sourcegraph-configured provider config for the rest.
+
+Example configuration
+
+{/*TODO: Missing code?*/}
```

Comment:
@taras-yemets looks like we have missing code here as well.

Reply:
Addressed in https://github.com/sourcegraph/docs/pull/735/commits/af56933b1beb98ae5969ce8c715f40eb3fbc07a6